packages/shared/profiles/ProfileAsPromise.ts

```diff
@@ -1,25 +1,28 @@
-import { getProfile, getProfileStatusAndData } from './selectors'
-import { profileRequest } from './actions'
+import { getProfile, getProfileStatusAndData, isAddedToCatalog } from './selectors'
+import { profileRequest, profileSuccess } from './actions'
 import { ProfileType } from './types'
 import { COMMS_PROFILE_TIMEOUT } from 'config'
 import { store } from 'shared/store/isolatedStore'
 import { Avatar } from '@dcl/schemas'
+import { isCurrentUserId } from 'shared/session/selectors'
 
 // We resolve a profile with an older version after this time, if there is that info
 const PROFILE_SOFT_TIMEOUT_MS = 5000
 
 // We reject the profile promise if more time than this has passed
 const PROFILE_HARD_TIMEOUT_MS = COMMS_PROFILE_TIMEOUT + 20000
 
-export function ProfileAsPromise(userId: string, version?: number, profileType?: ProfileType): Promise<Avatar> {
+// This method creates a promise that makes sure that a profile was downloaded AND added to renderer's catalog
+export async function ProfileAsPromise(userId: string, version?: number, profileType?: ProfileType): Promise<Avatar> {
   function isExpectedVersion(aProfile: Avatar) {
     return !version || aProfile.version >= version
   }
 
   const [, existingProfile] = getProfileStatusAndData(store.getState(), userId)
   const existingProfileWithCorrectVersion = existingProfile && isExpectedVersion(existingProfile)
-  if (existingProfile && existingProfileWithCorrectVersion) {
-    return Promise.resolve(existingProfile)
+  if (existingProfile && existingProfileWithCorrectVersion && !isCurrentUserId(store.getState(), userId)) {
+    store.dispatch(profileSuccess(existingProfile))
```

Comment:
in which case this line is necessary?

Reply:
This makes sure we go through the process posterior to a profile being fetched, particularly forwarding it to the renderer.

Reply:
in which cases does a profile exist locally and has not been sent to the renderer? I'm afraid that if we merge this line the next refactor removes it and breaks an use case

Reply:
I think the new version is what we are actually trying to do here, not trigger a new action, but rather making sure that the profile is added to the catalog before this promise resolves.

If we do not want this behavior, please, let's sync and find an alternative solution.